source/index.html.md

```diff
@@ -287,16 +514,103 @@ When you change to a new video (in the same player) you need to update the infor
+</p>
+
+<p class="lang-specific videojs jwplayer bitmovin ooyala">
+This is done by emitting a <code>videochange</code> event through <code>mux</code> which has been added to the <code>player</code> object, and when this is emitted it removes all previous video data and resets all metrics for the video view. See <a href="#metadata">Metadata</a> for the list of video details you can provide. You can include any metadata when changing the video but you should only need to update the values that start with <code>video_</code>.
 </p>
 
-<p class="lang-specific obj-c">
+<p class="lang-specific objective-c">
 This is done by calling <code>videoChangeForPlayer:withVideoData:</code> which will remove all previous video data and reset all metrics for the video view. See <a href="#metadata">Metadata</a> for the list of video details you can provide. You can include any metadata when changing the video but you should only need to update the values that start with <code>video_</code>.
 </p>
 
+## Sending Custom Error Events
+
+```video-element
+mux.emit('#myVideo', 'error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```videojs
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```jwplayer
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```bitmovin
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+```ooyala
+player.mux.emit('error', {
+  error_code: 100,
+  error_message: 'Description of error'
+});
+```
+
+<p class="lang-specific video-element videojs jwplayer bitmovin ooyala">
+Mux currently tracks fatal errors that are triggered by the player in use. However, in some cases, such as third-party HLS libraries for web players, fatal errors may occur outside of the context of the player. To handle these cases, Mux exposes a type of event that can be triggered via javascript. <strong>This method should only be used for fatal errors. Do not use this for "errors" that do result in playback failure</strong>
+<br><br>
+When triggering an error event, it is important to provide an <code>error_code</code> and <code>error_message</code> in an object as the second parameter. The <code>error_message</code> should provide a detailed description of the error as it happened, whereas the <code>error_code</code> must be an integer, and should provide a category of the error. If the errors match up with the <a href="https://developer.mozilla.org/en-US/docs/Web/API/MediaError">HTML Media Element Error</a>, you can use the same codes as the corresponding HTML errors. However, for custom errors, you should choose a number greater than or equal to 100. Whether all errors use the same code or you use different codes for different types of errors is up to you, but in general you should not send a distinct code for each possible error message.
```

Comment:
I really meant should not - no need to send 100, 101, 102, 103, etc... the codes should be more categorizations than distinct errors.